tensorbay/client/job.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+
+"""Basic structures of asynchronous jobs."""
+
+from typing import Any, Dict, Optional
+
+
+class GetLogResponse:
+    """This class defines the structure of the response from interface get_log."""
+
+    def __init__(self, logs: str) -> None:
+        self._logs = logs
+
+    def _get_logs(self, size: int = 100) -> str:
+        """Get some logs with limited size.
+
+        Arguments:
+            size: The limited size of log.
+
+        Return:
+            A string contains some logs.
+
+        """
+
+    def get_logs(self) -> str:
+        """Get some logs.
+
+        Return:
+            A string contains some logs.
+
+        """
+
+    def get_all_logs(self) -> str:
+        """Get all logs.
+
+        Return:
+            A string contains all logs.
+
+        """
+
+
+class Job:  # pylint: disable=too-many-instance-attributes
+    """This class defines :class:`Job`.
+
+    Arguments:
+        title: Title of the Job.
+        job_id: ID of the Job.
+        job_type: Type of the Job.
+        creator: The creator of the Job.
+        arguments: Arguments of the Job.
+        created_at: The time when the Job is created.
+        started_at: The time when the Job is started.
+        max_retries: The maximum retry times of the Job.
+        finished_at: The time when the Job is finished.
+        description: The description of the Job.
+
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        title: str,
+        job_id: str,
+        job_type: str,
+        creator: str,
+        arguments: Dict[str, Any],
+        created_at: str,
+        started_at: str,
+        max_retries: int,
+        finished_at: Optional[str] = None,
+        description: Optional[str] = "",
+    ) -> None:
+        self.title = title
+        self.job_id = job_id
+        self.job_type = job_type
+        self.creator = creator
+        self.arguments = arguments
+        self.created_at = created_at
+        self.started_at = started_at
+        self.max_retries = max_retries
+        self.finished_at = finished_at
```

Comment:
If the job not finish, the `finished_at` is None. Is that means: 
if I create a `Job` instance before finishing. Then the Job finished. how the `self.finished_at` be updated?

Reply:
Yes, the `self.finished_at ` will not be updated before `get_squash_and_merge_job` again.

Reply:
If so, shall we update `finished_at` every time user calling `job.finished_at` to keep `finished_at` latest.

Reply:
Yes, that makes more sense. But we need to use `get_job` to update 'finished_at'.

Reply:
I think is not a problem to use `get_job` to update `finish_at`.
How about the API and SDK in other platform?